source/views/menus/menu-bonapp.js

```diff
@@ -83,19 +86,28 @@ export class BonAppHostedMenu extends React.PureComponent<Props, State> {
     }
   }
 
+  requestMenu = (cafeId: string) => () =>
+    fetchJsonQuery(bonappMenuBaseUrl, {cafe: cafeId})
+  requestCafe = (cafeId: string) => () =>
+    fetchJsonQuery(bonappCafeBaseUrl, {cafe: cafeId})
+
   fetchData = async (props: Props) => {
     let cafeMenu: ?MenuInfoType = null
     let cafeInfo: ?CafeInfoType = null
 
     try {
       ;[cafeMenu, cafeInfo] = await Promise.all([
-        fetchJsonQuery(bonappMenuBaseUrl, {cafe: props.cafeId}),
-        fetchJsonQuery(bonappCafeBaseUrl, {cafe: props.cafeId}),
+        retry(this.requestMenu(props.cafeId), {retries: 3}),
```

Comment:
Is there a way for us to present a button of sorts to allow them to physically push "retry" instead of expecting them to switch the tab twice?

Reply:
… I was thinking they'd just go back to the homescreen.

Anyway. I (a) don't want a button because that's a bigger undertaking than I wanted to do here, and (b) because if we've tried three times and it's still failing, then I think we'll have other problems (the backoff starts at 1000ms).

So I don't think a button would help, and I don't really want to put it in right now. 😸

Reply:
My beef is that you've instructed them to try again.

Reply:
Fair. I still don't think that a button is beneficial here, given the error conditions in which I expect this screen to appear.

Perhaps I will come around later. Who knows.

Reply:
so uh… `NoticeView` supports a button and callback already. I'll … just go use that.

Reply:
I think that was what I had in mind and was hinting towards 😄